Sources/MapboxCoreNavigation/NativeHandlersFactory.swift

```diff
@@ -99,8 +108,26 @@ class NativeHandlersFactory {
                     endpointConfig: endpointConfig)
     }()
     
+    lazy var navigatorConfig: NavigatorConfig = {
+        return NavigatorConfig(voiceInstructionThreshold: nil,
+                               electronicHorizonOptions: nil,
+                               polling: nil,
+                               incidentsOptions: nil,
+                               noSignalSimulationEnabled: nil,
+                               avoidManeuverSeconds: NSNumber(value: RerouteController.DefaultManeuverAvoidanceRadius),
+                               useSensors: false)
+    }()
+
     lazy var configHandle: ConfigHandle = {
-        let customConfig = UserDefaults.standard.dictionary(forKey: customConfigKey) ?? [:]        
+        let nativeRerouterConfig = [
+            customConfigFeaturesKey: [
+                "useInternalReroute": true
+            ]
+        ]
+        
+        var customConfig = UserDefaults.standard.dictionary(forKey: customConfigKey) ?? [:]
+        customConfig.deepMerge(with: nativeRerouterConfig, uniquingKeysWith: { first, _ in first })
```

Comment:
Interesting choice to do a deep merge here. I guess I would’ve intuitively gone for getting `customConfigFeaturesKey` and directly setting its `useInternalReroute` to `true`, but I suppose the deep merge will scale better when we have more defaults to set.

Reply:
This custom config is a temporary requirement, while NN haven't enabled this feature by default. They will not do it until both iOS and Android integrate the feature, so we'll be forced to remove this "workaround" after that. I didn't put too much effort in designing this mechanism, I've just restored the code we had previously.